framework/include/executioners/SolveObject.h

```diff
@@ -41,7 +41,9 @@ class SolveObject : public MooseObject, public PerfGraphInterface, public Postpr
   /// Reference to FEProblem
   FEProblemBase & _problem;
   /// Displaced problem
-  std::shared_ptr<DisplacedProblem> _displaced_problem;
+  /// We would rather have a shared pointer here. But the displaced problem currently must be destructed
+  /// before the displaced mesh, and keeping a shared pointer here delays this
+  DisplacedProblem * _displaced_problem;
```

Comment:
So I guess only this move to raw pointer is the "dirtiness".

Reply:
it s kind of being endorsed in that discussion actually.
Alex says use less shared_ptr

Reply:
Oh sorry I misunderstood, I thought "my current solution" meant the changes you were proposing to make in libmesh, which would avoid the changes made here.